app/cni/pkg/install/validation.go

```diff
@@ -2,93 +2,58 @@ package install
 
 import (
 	"github.com/pkg/errors"
-
-	"github.com/kumahq/kuma/pkg/util/files"
 )
 
 func lookForValidConfig(files []string, checkerFn func(string) error) (string, bool) {
 	for _, file := range files {
-		err := checkerFn(file)
-		if err != nil {
+		if err := checkerFn(file); err != nil {
 			log.Info("error occurred testing config file", "file", file)
-		} else {
-			return file, true
+			continue
 		}
+
+		return file, true
 	}
+
 	return "", false
 }
 
 func isValidConfFile(file string) error {
 	parsed, err := parseFileToHashMap(file)
 	if err != nil {
-		return errors.Wrap(err, "could not unmarshal conf file")
+		return errors.Wrap(err, "failed to parse configuration file")
 	}
 
-	configType, ok := parsed["type"]
-	if ok {
-		log.V(1).Info("config valid", "file", file, "type", configType)
+	if configType, ok := parsed["type"]; ok {
+		log.V(1).Info("configuration validated", "file", file, "type", configType)
 		return nil
 	}
-	return errors.Errorf(`config file %v not valid - does not contain "type" field`, file)
+
+	return errors.Errorf(`configuration file "%s" missing "type" field`, file)
 }
 
 func isValidConflistFile(file string) error {
 	parsed, err := parseFileToHashMap(file)
 	if err != nil {
-		return errors.Wrap(err, "could not unmarshal conflist file")
+		return errors.Wrap(err, "failed to parse conflist file")
 	}
 
-	configName, hasName := parsed["name"]
-	plugins, hasPlugins := parsed["plugins"]
-	if hasName && hasPlugins {
-		log.V(1).Info("config valid", "file", file, "name", configName, "plugins", plugins)
-		return nil
-	}
-
-	return errors.Errorf(`config file %v not valid - does not contain "name" and "plugin" fields`, file)
-}
+	var missingFields []string
 
-func checkInstall(cniConfPath string, isPluginChained bool) error {
-	if !files.FileExists(cniConfPath) {
-		return errors.New("cni config file does not exist")
+	configName, ok := parsed["name"]
+	if !ok {
+		missingFields = append(missingFields, "name")
 	}
 
-	parsed, err := parseFileToHashMap(cniConfPath)
-	if err != nil {
-		return err
+	plugins, ok := parsed["plugins"]
+	if !ok {
+		missingFields = append(missingFields, "plugins")
 	}
 
-	if isPluginChained {
```

Comment:
Is this block deleted from the new version?

Reply:
No, the whole function was transformed to a method: https://github.com/kumahq/kuma/pull/12590/files/d433ae371ceb2baf56ab9ca2a187e4c09e4220fe#diff-4acb401431435486faf9ee1371e0580938c9dc6cc63720fc594b3cbb7e5dde38R191